Lib9c/Action/RegisterProduct.cs

```diff
@@ -0,0 +1,241 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Nekoyume.Helper;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.Market;
+using Nekoyume.Model.State;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    [ActionType("register_product")]
+    public class RegisterProduct : GameAction
+    {
+        public IEnumerable<IRegisterInfo> RegisterInfos;
```

Comment:
i think it can be check avatar address distinct count check https://github.com/planetarium/lib9c/pull/1733/files#diff-7dfbd7fe2b73e121e06d59b4c51a6b35ad6df442dd5729ef0fbfb08148c28494R30-R33